gadgets/fsnotify/test/unit/fsnotify_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	ebpftypes "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf/types"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/gadgetrunner"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+)
+
+type ExpectedFsnotifyEvent struct {
+	Timestamp string `json:"timestamp"`
+
+	Type string `json:"type"`
+
+	TraceeProc ebpftypes.Process `json:"tracee_proc"`
+	TracerProc ebpftypes.Process `json:"tracer_proc"`
```

Comment:
The process struct used in fsnotify is a bit different from `ebpftypes.Process`. I think it's better to define it here instead of using `ebpftypes.Process`

Reply:
Thank you so much @mauriciovasquezbernal , for the review!
I faced some challenges while debugging, so it took me a bit more time.

I didn't notice the difference at first, which might be why I wasn't getting any values in the process.
I have updated it with the defined struct.

However, I'm still not able to get values for `comm` and `pcomm`.
Let me know if I'm missing anything here.